controller/commentsController.js

```diff
@@ -51,3 +52,46 @@ exports.flagComment = async (req, res) => {
     errHandler(error, res);
   }
 };
+
+exports.deleteComment = async (req, res, next) => {
+  const commentId = req.params.commentId;
+  const userEmail = req.body.commentOwnerEmail;
+  try {
+    const comment = await Comments.findOne({ _id: commentId }).populate(
+      "commentOwner"
+    );
+    if (!comment) {
+      return next(new CustomError(400, "Comment not found"));
+    }
+    const email = await comment;
+    if (email.commentOwner.email.toLowerCase() == userEmail.toLowerCase()) {
```

Comment:
Update this to reflect the change requested above.
```suggestion
    if (comment.commentOwnerEmail.toLowerCase() == userEmail.toLowerCase()) {
```